app/helpers/course/controller_helper.rb

```diff
@@ -19,13 +19,12 @@ def display_course_user(user)
   # @return [String] The user-visible string, including embedded HTML which will display the
   #   string within a link to bring to the User page.
   def link_to_course_user(user, options = {})
-    link_path = course_user_path(user.course, user)
-    link_to(link_path, options) do
-      if block_given?
-        yield(user)
-      else
-        display_course_user(user)
-      end
+    link_text = capture { block_given? ? yield(user) : display_course_user(user) }
+    if user.approved?
+      link_path = course_user_path(user.course, user)
+      link_to(link_path, options) { link_text }
```

Comment:
In this circumstance then using the string form of link_to is better? `link_to(link_path, link_text, options)`?